main.tf

```diff
@@ -1,66 +1,87 @@
 terraform {
-  backend "remote" {
-    organization = "hashicorp-learn"
+  # backend "remote" {
+  #   organization = "hashicorp-learn"
 
-    workspaces {
-      name = "learn-terraform-pipelines-vault"
-    }
-  }
+  #   workspaces {
+  #     name = "learn-terraform-pipelines-vault"
+  #   }
+  # }
   required_providers {
     helm = {
       source  = "hashicorp/helm"
-      version = "~> 2.0.2"
+      version = "2.1.0"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "2.0.3"
     }
   }
 
   required_version = "~> 0.14"
 }
 
-data "terraform_remote_state" "cluster" {
-  backend = "remote"
+# data "terraform_remote_state" "cluster" {
+#   backend = "remote"
+#   config = {
+#     organization = var.organization
+#     workspaces = {
+#       name = var.cluster_workspace
+#     }
+#   }
+# }
+
+# data "terraform_remote_state" "consul" {
+#   backend = "remote"
+#   config = {
+#     organization = var.organization
+#     workspaces = {
+#       name = var.consul_workspace
+#     }
+#   }
+# }
+
+data "terraform_remote_state" "eks_cluster" {
+  backend = "local"
   config = {
-    organization = var.organization
-    workspaces = {
-      name = var.cluster_workspace
-    }
+    path = "../learn-terraform-pipelines-k8s/terraform.tfstate"
   }
 }
 
 data "terraform_remote_state" "consul" {
-  backend = "remote"
+  backend = "local"
   config = {
-    organization = var.organization
-    workspaces = {
-      name = var.consul_workspace
-    }
+    path = "../learn-terraform-pipelines-consul/terraform.tfstate"
   }
 }
 
-
-# Retrieve GKE cluster information
-provider "google" {
-  project = data.terraform_remote_state.cluster.outputs.project_id
-  region  = data.terraform_remote_state.cluster.outputs.region
-}
-
-data "google_client_config" "default" {}
-
-data "google_container_cluster" "my_cluster" {
-  name     = data.terraform_remote_state.cluster.outputs.cluster
-  location = data.terraform_remote_state.cluster.outputs.region
-}
-
 provider "kubernetes" {
-  host                   = data.terraform_remote_state.cluster.outputs.host
-  token                  = data.google_client_config.default.access_token
-  cluster_ca_certificate = data.terraform_remote_state.cluster.outputs.cluster_ca_certificate
-
+  host                   = data.terraform_remote_state.eks_cluster.outputs.endpoint
+  cluster_ca_certificate = data.terraform_remote_state.eks_cluster.outputs.cluster_ca_certificate
+  exec {
+    api_version = "client.authentication.k8s.io/v1alpha1"
+    command     = "aws"
+    args = [
+      "eks",
+      "get-token",
+      "--cluster-name",
+      data.terraform_remote_state.eks_cluster.outputs.cluster
+    ]
+  }  
```

Comment:
Update to use [`eks_cluster_auth`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/eks_cluster_auth) instead since the AWS CLI prob wont be configured in the Katacoda scenario